allennlp/data/tokenizers/basic.py

```diff
@@ -0,0 +1,33 @@
+from typing import List
+
+from overrides import overrides
+
+from allennlp.common import Params
+from allennlp.data.tokenizers.token import Token
+from allennlp.data.tokenizers.tokenizer import Tokenizer
+
+@Tokenizer.register("basic")
+class BasicTokenizer(Tokenizer):
+    """
+    A ``BasicTokenizer`` splits strings of text into tokens by treating space characters as 
+    separator, and pad a special EOF token to the end of each sentence. This tokenizer is 
+    relative fast and only provides the very basic function as a tokenizer (suitable for
+    large, already tokenized corpora, e.g., the wiki103 dataset).
+    """
+
+    @overrides
+    def tokenize(self, text: str) -> List[Token]:
+        if text and not text.isspace():
+            return [Token(word) for word in text.split()] + [Token("<eof>")]
```

Comment:
We already have `</S>` for the ELMo LM end-of-sentence token, so using `<eof>` here will get confusing.  Is it possible to handle the end-of-sentence token here in the same manner as the existing ELMo code?

Reply:
Sure, no problem :-)